### ISSUE TITLE
Bug: no-async-array-callbacks false negative for deferred Promise.all

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#30]
## Description

The `isSafelyConsumedMapResult` function in `no-async-array-callbacks.ts` only considers cases where the `Promise.all` or `Promise.allSettled` is immediately awaited or returned. It doesn't handle the common pattern where the map result is stored in a variable and awaited later:

```ts
// This is safe but would be flagged incorrectly
const promises = items.map(async (item) => processItem(item));
await Promise.all(promises);
```

## Expected Behavior

This pattern should be considered safe since the promises are properly awaited.

## Actual Behavior

The rule flags the async map callback as unsafe.

## Root Cause

`isSafelyConsumedMapResult` only checks if the map call is directly passed to `Promise.all`, not if it's stored and later consumed.

## Suggested Fix

Consider tracking variable declarations that hold map results and check if they're later passed to `Promise.all`/`Promise.allSettled` and awaited.
